component/background/StorageService.tsx

```diff
@@ -66,6 +96,10 @@ class StorageService {
    * @return {Promise<{}>} A promise whose resolve takes no argument, which runs after the key is removed
    */
   public remove(key: string): Promise<{}> {
+    if (!this.status) {
+      this.initialize();
```

Comment:
Isn't initialized async, so it means it does not really block anything. You probably want to chain the promises.